presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -819,6 +822,14 @@ private static OptionalLong parse(@Nullable String parameterValue)
         statistics.getInMemoryDataSizeInBytes().ifPresent(size -> result.put(RAW_DATA_SIZE, Long.toString(size)));
         statistics.getOnDiskDataSizeInBytes().ifPresent(size -> result.put(TOTAL_SIZE, Long.toString(size)));
 
+        if (hiveFlavor == HiveFlavor.UNKNOWN || hiveFlavor == HiveFlavor.CDH5) {
+            // CDH 5.16 metastore ignores stats unless STATS_GENERATED_VIA_STATS_TASK is set
+            // https://github.com/cloudera/hive/blob/cdh5.16.2-release/metastore/src/java/org/apache/hadoop/hive/metastore/MetaStoreUtils.java#L227-L231
+            if (!parameters.containsKey("STATS_GENERATED_VIA_STATS_TASK")) {
+                result.put("STATS_GENERATED_VIA_STATS_TASK", "workaround for potential lack of HIVE-12730");
```

Comment:
What if we set this for other metastores (like for HDP3)?

Reply:
We can and we do (by default).
The config allows one to disable what can be seen as littering.

Reply:
```
result.put("STATS_GENERATED_VIA_STATS_TASK", "Presto");
```
Does not look as litter.

Reply:
@electrum wanted the property value to be more self-explanatory. I can add "Presto" to it, but still want to retain "HIVE-12730" (with some additional words) there.

Reply:
My point is to do it unconditionally (if does not introduce regression) and remove `HiveFlavor`

Reply:
i talked about this config with @electrum and we wanted to have it.
Of course, we can revert this decision if we all agree to do so.

Reply:
I agree that flag could be useful and maybe at some point we would need it.

Today I don't think having it just to set a parameter for stats is enough to justify it existence.

Having this configuration creates a notion that there is a bigger influence on Presto about how it communicates with HMS. While in reality it is very minor. And between values `UNKNOWN,  AWS_GLUE,  FILE_METASTORE, HDP2, HDP3,     AWS_EMR, CDH6` there is no difference.

I would like to hear @electrum opinion about this.